ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -733,16 +1147,29 @@ def _checkConvObjective(self, traj, **kwargs):
       @ In, kwargs, dict, dictionary of parameters for convergence criteria
       @ Out, converged, bool, convergence state
     """
+    # _optPointHistory is used to check that we run the algorithm one step
     if len(self._optPointHistory[traj]) < 2:
       return False
-    o1, _ = self._optPointHistory[traj][-1]
-    obj = o1[self._objectiveVar]
-    converged = (obj == self._convergenceCriteria['objective'])
-    self.raiseADebug(self.convFormat.format(name='objective',
-                                            conv=str(converged),
-                                            got=obj,
-                                            req=self._convergenceCriteria['objective']))
-
+    # An alternative was to use:
+    # o1, _ = self._optPointHistory[traj][-1]
+    # but that will only search the "best points" so is slower at finding
+    # one that matches the objective.
+    o1 = kwargs['new']
+    #print(f"{o1=}")
+    for j in range(len(np.atleast_1d(o1[self._objectiveVar[0]]))):
+      converged = True
+      bestObjective = []
+      for i,objVar in enumerate(self._objectiveVar):
+        #print(f"{objVar=}  {np.atleast_1d(o1[objVar])[j]=} {self._convergenceCriteria['objective'][i]=}")
+        currentObj = np.atleast_1d(o1[objVar])[j]*self._objMult[objVar]
+        bestObjective.append(currentObj*self._objMult[objVar])
```

Comment:
Here is another place to transform the multi-objective value, Let's try to make sure this is the right place to do it, and also let's use function to do it and I hope that will help us to better track all these transformations.

Reply:
As I pointed out in a different comment, using a function to do this would complicate the code (I would probably need to define a temporary variable (or two), then use the function to transform that, then take the temporary variable and put it in the final variable), so I am not sure that changing this would be an improvement.